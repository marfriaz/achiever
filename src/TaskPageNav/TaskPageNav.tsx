import React, { useContext } from "react";
import { useParams, useHistory } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import CircleButton from "../CircleButton/CircleButton";
import ApiContext from "../ApiContext";
import { findTask, findGoal } from "../tasks-helpers";
import "./TaskPageNav.css";

interface ParamTypes {
  taskId: string;
}
export default function TaskPageNav() {
  const history = useHistory();

  const { tasks, goals } = useContext(ApiContext)!;
  const { taskId } = useParams<ParamTypes>();
  const task = findTask(tasks, taskId) || { goal_id: undefined };
  const goal = findGoal(goals, task.goal_id);
  return (
    <div className="TaskPageNav">
      <CircleButton
        tag="button"
        onClick={() => history.goBack()}
        className="TaskPageNav__back-button"
      >
        <FontAwesomeIcon icon="chevron-left" />
        <br />
        Back
      </CircleButton>
      {goal && <h3 className="TaskPageNav__goal-name">{goal.name}</h3>}
    </div>
  );
}
