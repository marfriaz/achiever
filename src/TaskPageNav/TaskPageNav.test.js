import React from "react";
import { shallow } from "../GoalListNav/node_modules/enzyme";
import toJson from "../GoalListNav/node_modules/enzyme-to-json";
import TaskPageNav from "./TaskPageNav";

describe(`TaskPageNav component`, () => {
  it("renders a .TaskPageNav by default", () => {
    const wrapper = shallow(<TaskPageNav />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // enzyme doesn't support React.createContext
  it.skip("renders a h3 with goal name when in props", () => {
    const props = {
      match: {
        params: {
          taskId: "test-task-id",
        },
      },
    };
    const context = {
      tasks: [{ id: "test-task-id", goalId: "test-goal-id" }],
      goals: [{ id: "test-goal-id", name: "Important" }],
    };

    const h3 = shallow(<TaskPageNav {...props} />, context).find(
      ".TaskPageNav__goal-name"
    );
    expect(toJson(h3)).toMatchSnapshot();
  });
});
