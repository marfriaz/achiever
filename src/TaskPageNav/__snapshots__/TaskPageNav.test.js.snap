// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskPageNav component renders a .TaskPageNav by default 1`] = `
<div
  className="TaskPageNav"
>
  <NavCircleButton
    className="TaskPageNav__back-button"
    onClick={[Function]}
    role="link"
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="chevron-left"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <br />
    Back
  </NavCircleButton>
</div>
`;

exports[`TaskPageNav component renders a h3 with goal name when in props 1`] = `
<h3
  className="TaskPageNav__goal-name"
>
  Important
</h3>
`;
